moveit_core/trajectory_processing/src/time_optimal_trajectory_generation.cpp

```diff
@@ -981,6 +1004,40 @@ bool TimeOptimalTrajectoryGeneration::computeTimeStamps(
     RCLCPP_ERROR(LOGGER, "It looks like the planner did not set the group the plan was computed for");
     return false;
   }
+
+  // Validate scaling
+  double velocity_scaling_factor = 1.0;
+  if (max_velocity_scaling_factor > 0.0 && max_velocity_scaling_factor <= 1.0)
+  {
+    velocity_scaling_factor = max_velocity_scaling_factor;
+  }
+  else if (max_velocity_scaling_factor == 0.0)
+  {
+    RCLCPP_DEBUG(LOGGER, "A max_velocity_scaling_factor of 0.0 was specified, defaulting to %f instead.",
```

Comment:
Nit (Optional): Since this code occurs in 4 places with these changes, do you think it would be much work to create a single (free?) function that applies these checks? Something like:

```
bool verifyScalingFactor(double& scaling_factor, double max_scaling_factor){
   ...
}
```

Reply:
Haha yeah I probably should have done that to begin with. I'll make the helper function